seasmon_xr/ops/zonal.py

```diff
@@ -0,0 +1,43 @@
+"""Numba accelerated zonal statistics."""
+from numba import njit
+from numpy import zeros
+
+from ._helper import lazycompile
+
+
+@lazycompile(njit)
+def do_mean(pixels, z_pixels, nodata, num_zones, dtype="float64"):
+    """Calculate the zonal mean.
+
+    The mean for each pixel of `pixels` is calculated for each zone in
+    `z_pixels`.
+
+    The zones in `z_pixels` have to be numbered in a linear fashion,
+    starting with 0 for the first zone and num_zones for the last zone.
+
+    Args:
+        pixels: input value pixels (T,Y,X)
+        z_pixels: zonal pixels (Y,X)
+        nodata: nodata value in values
+        num_zones: number of zones in z_pixels
+        dtype: datatype
+    """
+    t, nr, nc = pixels.shape
+
+    sums = zeros((t, num_zones), dtype=dtype)
+    valids = zeros((t, num_zones), dtype=dtype)
+
+    for tix in range(t):
+        for rw in range(nr):
+            for cl in range(nc):
+                pix = pixels[tix, rw, cl]
+                z_idx = z_pixels[rw, cl]
+                if pix != nodata and z_idx >= 0:
+                    sums[tix, z_idx] += pix
+                    valids[tix, z_idx] += 1
+
+        for idx in range(sums.shape[1]):
+            if valids[tix, idx] > 0:
+                sums[tix, idx] = sums[tix, idx] / valids[tix, idx]
+
+    return sums
```

Comment:
It would actually be nice if we could also return `valids` - I don't think that would have any impact besides having to change the accessor code that calls the function and puts the return into an `xarray` object. I did try that a couple of times but always got tripped up on getting dimensions right in `map_blocks`

Reply:
Easiest is to add another dimension to the output. I'm still not sure what is the best way to code "map blocks like" dask operations with multiple array outputs.